app/assets/javascripts/discourse/initializers/topic-footer-buttons.js.es6

```diff
@@ -0,0 +1,116 @@
+      });
+
+      api.registerTopicFooterButton({
+        id: "flag",
+        icon: "flag",
+        priority: 998,
+        label: "topic.flag_topic.title",
+        title: "topic.flag_topic.help",
+        action: "showFlagTopic",
+        dropdown: context => context.site.mobileView,
+        class: "flag-topic",
+        dependentKeys: [
+          "topic.details.can_flag_topic",
+          "topic.isPrivateMessage"
+        ],
+        displayed: context =>
+          context.get("topic.details.can_flag_topic") &&
+          !context.get("topic.isPrivateMessage")
+      });
+
+      api.registerTopicFooterButton({
+        id: "invite",
+        icon: "users",
+        priority: 997,
+        label: "topic.invite_reply.help",
+        title: "topic.invite_reply.help",
+        action: "showInvite",
+        dropdown: context => context.site.mobileView,
+        class: "invite-topic",
+        dependentKeys: ["canInviteTo", "inviteDisabled"],
+        displayed: context => context.get("canInviteTo"),
+        disabled: context => context.get("inviteDisabled")
+      });
+
+      api.registerTopicFooterButton({
+        dependentKeys: ["topic.bookmarked", "topic.isPrivateMessage"],
+        id: "bookmark",
+        icon: "bookmark",
+        priority: 1000,
+        class: context => {
+          const bookmarked = context.get("topic.bookmarked");
+          return bookmarked ? "bookmark bookmarked" : "bookmark";
+        },
+        label: context => {
+          const bookmarked = context.get("topic.bookmarked");
+          return bookmarked ? "bookmarked.clear_bookmarks" : "bookmarked.title";
+        },
+        title: context => {
+          const bookmarked = context.get("topic.bookmarked");
+          return bookmarked
+            ? "bookmarked.help.unbookmark"
+            : "bookmarked.help.bookmark";
+        },
+        action: "toggleBookmark",
+        dropdown: context => context.site.mobileView,
+        displayed: context => !context.get("topic.isPrivateMessage")
+      });
+
+      api.registerTopicFooterButton({
+        id: "archive",
+        priority: 1001,
+        icon: context => context.get("archiveIcon"),
+        label: context => context.get("archiveLabel"),
+        title: context => context.get("archiveTitle"),
+        action: context => context.get("toggleArchiveMessage"),
+        dropdown: true,
+        class: "standard archive-topic",
+        dependentKeys: [
+          "canArchive",
+          "archiveIcon",
+          "archiveLabel",
+          "archiveTitle",
+          "toggleArchiveMessage"
+        ],
+        displayed: context => context.get("canArchive")
+      });
+
+      api.registerTopicFooterButton({
+        id: "edit-message",
+        priority: 750,
+        icon: "pencil-alt",
+        label: "topic.edit_message.title",
+        title: "topic.edit_message.help",
+        action: context => context.get("editFirstPost"),
+        dropdown: false,
```

Comment:
I would assume the default is no dropdown and that this would be unnecessary.

Reply:
Yep was thinking the same :+1: